R/drive_download.R

```diff
@@ -43,49 +43,104 @@ drive_download <- function(file = NULL,
   file <- as_dribble(file)
   file <- confirm_single_file(file)
 
+  ## preserve extension from out_path, before possible override by file$name
+  ext <- file_ext_safe(out_path)
```

Comment:
I think we only want to consult the extension if it came directly from `out_path`, not if it's from `file$name`. Why? One example is the upload of `foo.csv` into a Sheet. It will still be named `foo.csv` (a regrettable choice, yes, but possible). But we don't want this unfortunate name to drive the download file type, which should default to xlsx for a Sheet. We should discuss....